span_sink.go

```diff
@@ -296,7 +301,11 @@ func (ls *lightStepSpanSink) Name() string {
 
 // Ingest takes in a span and passed it along to the LS client after
 // some sanity checks and improvements are made.
-func (ls *lightStepSpanSink) Ingest(ssfSpan ssf.SSFSpan) error {
+func (ls *lightStepSpanSink) Ingest(ssfSpan *ssf.SSFSpan) error {
+	if err := protocol.ValidateTrace(ssfSpan); err != nil {
```

Comment:
Boilerplate code like this seems something we should try and elevate up into the worker or the parsing phase. If we have to do this in each sink's `Ingest` someone will inevitably forget. The contract could be "`Ingest` is only passed legit spans", right?

Reply:
So the thing is - these *are* legit SSF spans. They just might not contain any traceable material that the sink can handle. I guess I could rename the `ValidateTrace` function to be more reasonable.

Also, kinda agreed about boilerplate code, but then it's kinda not exactly boilerplate - as the metric extractor shows, there's an entirely reasonable use case for processing SSFSpans that aren't valid trace data (:

Reply:
Oh, interesting. The problem being that a _trace_ might not have a span in it?